[Doc] Realtime integration in Navitia (#2811)
CanalTP/navitia: documentation/slate/source/includes/realtime.md

<a name="realtime"></a> Real time integration in Navita
===============================================

> Several endpoints can integrate real time information in their responses. In the response received, a disruption will be present and additional information will be provided if the parameter `data_freshness` is set to `realtime`.

Author: Disruption is present even without 'data_freshness=realtime' for information. An other request has to be made to apply disruption.

<aside class="notice">

> Real time isn't available on every coverages for Navitia. For real time to be available for a client, it needs to provide real time info about its network to Navitia, which can be provided through several means (GTFS-RT for ex or proprietary format like the SNCF client).

Author: Remove the second part

</aside>

The effect of a disruption can be one of the following:
<ul>
	<li>SIGNIFICANT_DELAYS</li>
	<li>REDUCED_SERVICE</li>
</ul>

For each one of these effects, here's how the Navitia responses will be affected over the different endpoints. A link to the API

## Trip delayed

The effect of the disruption is `SIGNIFICANT_DELAYS`. It means that the train will arrive late at one or more stations in its journey.

```
{
    "amended_arrival_time": "093200",
    "amended_departure_time": "093300",
    "arrival_status": "delayed"
    "base_arrival_time": "092200",
    "base_departure_time": "092300",
    "cause": "Panne d'un aiguillage",
    "departure_status": "delayed",
    "is_detour": false,
    "stop_point": ⊕{7 items},
    "stop_time_effect": "delayed",
}
```

In the disruption, the delay can be found in the list of "impacted_stops" with the departure/arrival status set to "delayed". See the [disruption](#disruption) objects section for its full content and description.

> ### Journeys

Author: CSS with front-end dev?

Author: The power of div :smiling_imp:

Reviewer: What are you trying to do here? 
To reduce the size of the title I think there was some css (can search a bit more if you ask)

Author: No, to have the description and the shell of one endpoint facing each other. And it does that!

<a name="realtime"></a> Real time integration in Navita
===============================================

Several endpoints can integrate real time information in their responses. In the response received, a disruption will be present and additional information will be provided if the parameter `data_freshness` is set to `realtime`. If the parameter `data_freshness` is set to `base_schedule`, the disruption is also present in the response for the user information, but it won't be taken into account in the results of the query.

<aside class="warning">

> Real time isn't available on every coverages for Navitia. For real time to be available for a client, it needs to provide real time info about its network to Navitia.

Reviewer 2: ```suggestion
Real time isn't available on every coverage for Navitia. For real time to be available for a client, it needs to provide real time info about its network to Navitia.
```

</aside>

The effect of a disruption can be one of the following:
<ul>

> 	<li>[SIGNIFICANT_DELAYS](#SIGNIFICANT_DELAYS)</li>

Reviewer 2: minor: Can you add a simple diagram of the case (Like in the demo slides)

                    "base_departure_date_time": "20190529T091600"
                    "departure_date_time": "20190529T092600",
                    "links": [],
                    "stop_point": ⊕{7 items},
                },
               ⊖{
                    "additional_informations": [],
                    "arrival_date_time": "20190529T093200",
                    "base_arrival_date_time": "20190529T092200",
                    "base_departure_date_time": "20190529T092300"
                    "departure_date_time": "20190529T093300",
                    "links": [],
                    "stop_point": ⊕{7 items},
                },
               ⊖{
                    "additional_informations": [],
                    "arrival_date_time": "20190529T095100",
                    "base_arrival_date_time": "20190529T094100",
                    "base_departure_date_time": "20190529T094100"
                    "departure_date_time": "20190529T095100",
                    "links": [],
                    "stop_point": ⊕{7 items},
                }
            ]
            "to": ⊕{5 items},
            "type": "public_transport",
        }
    ]
```

> The status of the journey is `SIGNIFICANT_DELAYS`. In a public transport section of the response, "base_arrival_date_time"/"base_departure_date_time" represent the scheduled arrival/departure time without taking into account the delay whereas "arrival_date_time"/"departure_date_time" are the actual arrival/departure time, after the delay is applied. The delay can also be observed for every stop_point of the journey with th same parameters in "stop_date_times".

Reviewer 2: ```suggestion
The status of the journey is `SIGNIFICANT_DELAYS`. In a public transport section of the response, "base_arrival_date_time"/"base_departure_date_time" represent the scheduled arrival/departure time without taking into account the delay whereas "arrival_date_time"/"departure_date_time" are the actual arrival/departure time, after the delay is applied. The delay can also be observed for every stop_point of the journey with the same parameters in "stop_date_times".
```

<a name="realtime"></a> Real time integration in Navita
===============================================

Several endpoints can integrate real time information in their responses. In the response received, a disruption will be present and additional information will be provided if the parameter `data_freshness` is set to `realtime`. If the parameter `data_freshness` is set to `base_schedule`, the disruption is also present in the response for the user information, but it won't be taken into account in the results of the query.

<aside class="warning">
Real time isn't available on every coverage for Navitia. For real time to be available for a client, it needs to provide real time info about its network to Navitia.
</aside>

The effect of a disruption can be one of the following:

> <ul>

Reviewer: Should we add something about the `UNKNOWN_EFFECT` that means normal or back-to-normal ?

Author: There's no ticket for that! :wink:  :grin:  
Update: https://jira.kisio.org/browse/NAVP-1360

Author: Added in https://jira.kisio.org/browse/NAVP-1325

	<li>[SIGNIFICANT_DELAYS](#SIGNIFICANT_DELAYS)</li>
	<li>[REDUCED_SERVICE](#REDUCED_SERVICE)</li>
	<li>[MODIFIED_SERVICE](#MODIFIED_SERVICE)</li>
</ul>

> For each one of these effects, here's how the Navitia responses will be affected over the different endpoints. A link to the API

Reviewer: Looks like a wrong copy-paste here :
```suggestion
For each one of these effects, here's how the Navitia responses will be affected over the different endpoints.
```

Author: Indeed

## <a name="SIGNIFICANT_DELAYS"></a>Trip delayed

> ![image](delay.png)

Reviewer: :gb: : There "de retard" is :fr: in image provided.

minor: image is a bit too big to me (same on all).

Author: check with @ArthurLeyv

> ``` shell

Reviewer: minor : You may put this above image so that the shell example is facing the image.
Same on all images.

Author: The shell example should face the explanation (2 schools I guess)

# Extract of an impacted stop of /disruptions
{
    "amended_arrival_time": "194700",
    "amended_departure_time": "194900",
    "arrival_status": "delayed",
    "base_arrival_time": "193200",
    "base_departure_time": "193400",
    "cause": "Panne d'un aiguillage",
    "departure_status": "delayed",
    "is_detour": false,
    "stop_point": ⊕{7 items},
    "stop_time_effect": "delayed",
}
```
The effect of the disruption is `SIGNIFICANT_DELAYS`. It means that the train will arrive late at one or more stations in its journey.

In the disruption, the delay can be found in the list of "impacted_stops" with the departure/arrival status set to "delayed". See the [disruption](#disruption) objects section for its full content and description.

<div></div>
### Journeys

``` shell
# Request example for /journeys
http://api.navitia.io/v1/coverage/<coverage>/journeys?from=<origin>&to=<destination>&data_freshness=realtime
```

``` shell
# Extract of the public transport section of the response /journeys

    "sections": ⊖[
       ⊖{

>             "additional_informations": ⊕[1 item],

Reviewer: You could use a section that's really impacted doing a realtime call so that you can show:
* arrival and base_arrival are different
* display_information.links points to the internal disruption collection

```
            "additional_informations": ⊕[1 item],
            "arrival_date_time": "20190619T123200",
            "base_arrival_date_time": "20190619T122700",
            "base_departure_date_time": "20190619T113900",
            "co2_emission": ⊕{2 items},
            "data_freshness": "realtime",
            "departure_date_time": "20190619T114400",
            "display_informations": ⊖{
                "code": "",
                "color": "000000",
                "commercial_mode": "TER",
                "description": ""
                "direction": "Grenoble (Grenoble)",
                "equipments": [],
                "headsign": "17520",
                "label": "Geneve - Evian - Annecy - Valence - Avignon",
                "links": ⊖[
                   ⊖{
                        "id": "b59bdab8-3560-4cfe-8009-b0461f74c417",
                        "internal": true,
                        "rel": "disruptions",
                        "templated": false
                        "type": "disruption",
                    }
                ],
                "name": "Geneve - Evian - Annecy - Valence - Avignon",
                "network": "SNCF",
                "physical_mode": "Train régional / TER",
                "text_color": "",
            },
            "duration": 2880,
            "from": ⊕{5 items},
            "geojson": ⊕{3 items},
            "id": "section_1_0",
            "links": ⊕[6 items],
            "stop_date_times": ⊖[
               ⊖{
                    "additional_informations": [],
                    "arrival_date_time": "20190619T114100",
                    "base_arrival_date_time": "20190619T113600",
                    "base_departure_date_time": "20190619T113900"
                    "departure_date_time": "20190619T114400",
                    "links": [],
                    "stop_point": ⊕{7 items},
                },
               ⊕{7 items},
               ⊕{7 items},
               ⊕{7 items},
               ⊕{7 items}
            ]
            "to": ⊕{5 items},
            "type": "public_transport",
```
My call : https://api.sncf.com/v1/coverage/sncf/journeys?from=admin%3Afr%3A73065&to=stop_area%3AOCE%3ASA%3A87747006&data_freshness=realtime&datetime=20190619T113500&

Author: TO DO by @pbougue

            "arrival_date_time": "20190529T205000",
            "base_arrival_date_time": "20190529T204500",
            "base_departure_date_time": "20190529T160000",
            "co2_emission": ⊕{2 items},
            "data_freshness": "realtime",
            "departure_date_time": "20190529T160000",
            "display_informations": ⊕{13 items},
            "duration": 21180,
            "from": ⊕{5 items},
            "geojson": ⊕{3 items},
            "id": "section_0_0",
            "links": ⊕[6 items],
            "stop_date_times": ⊖[
               ⊕{7 items},
               ⊖{
                    "additional_informations": [],
                    "arrival_date_time": "20190605T194700",
                    "base_arrival_date_time": "20190605T193200",
                    "base_departure_date_time": "20190605T193400"
                    "departure_date_time": "20190605T194900",
                    "links": [],
                    "stop_point": ⊕{7 items},
                },
               ⊕{7 items},
               ⊕{7 items},
            ]
            "to": ⊕{5 items},
            "type": "public_transport",
        }
    ]
```

> The status of the journey is `SIGNIFICANT_DELAYS`. In a public transport section of the response, "base_arrival_date_time"/"base_departure_date_time" represent the scheduled arrival/departure time without taking into account the delay whereas "arrival_date_time"/"departure_date_time" are the actual arrival/departure time, after the delay is applied. The delay can also be observed for every stop_point of the journey with the same parameters in "stop_date_times".

Reviewer 3: "stop_point" is a value from the code so it should be between comma ?

Author: Why? Why not? Let's call a referee!

Reviewer: No big preference, but I'd vote to let it as-is.
To me, it's a transportation object in that context.
Maybe we can write `stop point` or `stop-point` to show that it's kind of real-life object, but the current way is probably the most consistent with the rest of the doc.

Author: :checkered_flag: `stop point` everywhere :champagne:

> The status of the journey is `SIGNIFICANT_DELAYS`. In a public transport section of the response, "base_arrival_date_time"/"base_departure_date_time" represent the scheduled arrival/departure time without taking into account the delay whereas "arrival_date_time"/"departure_date_time" are the actual arrival/departure time, after the delay is applied. The delay can also be observed for every stop_point of the journey with the same parameters in "stop_date_times".

Reviewer: minor: Some space can be added
```suggestion
The status of the journey is `SIGNIFICANT_DELAYS`.

In a public transport section of the response, "base_arrival_date_time"/"base_departure_date_time" represent the scheduled arrival/departure time without taking into account the delay whereas "arrival_date_time"/"departure_date_time" are the actual arrival/departure time, after the delay is applied.  
The delay can also be observed for every stop_point of the journey with the same parameters in "stop_date_times".
```

Author: ok

If the parameter "data_freshness" is set to "base_schedule",  "base_arrival_date_time"/"base_departure_date_time" = "arrival_date_time"/"departure_date_time".

> A list of the disruptions impacting the journey is also present at the root level of the response.

Reviewer: You may also add (if you change the example as suggested above):
```
In a disrupted section, `display_informations.links` contains an internal link to the concerned disruption.
```

Author: ok; TO DO by @pbougue with changes of request /journeys exemple

<aside class="notice">
For a disruption to be present in the response, the request has to be made during its application period.
</aside>

<div></div>
### Departures

``` shell

> # Request example for /departures

Reviewer: minor: In departures, default is `data_freshness=realtime`. Hope we don't have a bug on that.
We may document it like:
```suggestion
# Request example for /departures (data_freshness=realtime by default)
```

> http://api.navitia.io/v1/coverage/<coverage>/physical_modes/<physical_mode>/stop_points/<stop_point>/departures?from_datetime=<from_date>&data_freshness=realtime

Reviewer: Please switch the example call to stop_area (to be consistent with trip-add and stop_add when they will be documented)

Author: As discussed in refinement, indeed!

```

``` shell
# Extract of an impacted departure object of the response /departures

{

>     "display_informations": ⊕{13 items},

Reviewer: Here you can also develop display_informations to show the link:
```
    "display_informations": ⊖{
        "code": "",
        "color": "000000",
        "commercial_mode": "TER",
        "description": ""
        "direction": "Grenoble (Grenoble)",
        "equipments": [],
        "headsign": "17520",
        "label": "Geneve - Evian - Annecy - Valence - Avignon",
        "links": ⊖[
           ⊖{
                "id": "b59bdab8-3560-4cfe-8009-b0461f74c417",
                "internal": true,
                "rel": "disruptions",
                "templated": false
                "type": "disruption",
            }
        ],
        "name": "Geneve - Evian - Annecy - Valence - Avignon",
        "network": "SNCF",
        "physical_mode": "Train régional / TER",
        "text_color": "",
    },
    "links": ⊕[6 items],
    "route": ⊕{9 items},
    "stop_date_time": ⊖{
        "additional_informations": [],
        "arrival_date_time": "20190619T114100",
        "base_arrival_date_time": "20190619T113600",
        "base_departure_date_time": "20190619T113900",
        "data_freshness": "realtime"
        "departure_date_time": "20190619T114400",
        "links": [],
    }
    "stop_point": ⊕{11 items},
```
Call : https://api.sncf.com/v1/coverage/sncf/stop_areas/stop_area%3AOCE%3ASA%3A87741009/departures?from_datetime=20190619T113500&

Author: TO DO by @pbougue

    "links": ⊕[6 items],
    "route": ⊕{9 items},
    "stop_date_time": ⊖{
        "additional_informations": [],
        "arrival_date_time": "20190605T194700",
        "stop_date_time": "realtime"
        "departure_date_time": "20190605T194900",
        "links": [],
    }
    "stop_point": ⊕{11 items},
}
```

In the "stop_date_time" section of the response, the parameter "stop_date_time" is "realtime" and the fields "arrival_date_time"/"departure_date_time" take the delay into account.

> A list of the disruptions impacting the departures is also present at the root level of the response.

Reviewer: Same: you can add a word on `display_informations.links`

Author: TO DO by @pbougue with response exemple

<div></div>
### Stop Schedules

``` shell

> # Request example for /stop_schedules

Reviewer: minor: If you mention above that the default is realtime, we should probably mention here:
```suggestion
# Request example for /stop_schedules (data_freshness=realtime by default)
```
Do not forget to modify the doc : https://jira.kisio.org/browse/NAVP-1357

> http://api.navitia.io/v1/coverage/<coverage>/physical_modes/<physical_mode>/lines/<line>/stop_points/<stop_point>/stop_schedules?from_datetime=<from_date>&data_freshness=realtime

Reviewer: Please switch the example call to stop_area (to be consistent with trip-add and stop_add when they will be documented)

```

``` shell
# Extract of an impacted stop_schedules object of the response /stop_schedules

{
    "additional_informations": null,
    "date_times": ⊖[
       ⊕{5 items},
       ⊖{
            "additional_informations": [],
            "data_freshness": "realtime",
            "date_time": "20190605T194900",

>             "links": ⊕[2 items]

Reviewer: Here you can expand links and show that you're supposed to have a base_date_time:
```
    "additional_informations": null,
    "date_times": ⊖[
       ⊖{
            "additional_informations": [],
            "base_date_time": "20190619T113900",
            "data_freshness": "realtime",
            "date_time": "20190619T114400",
            "links": ⊖[
               ⊕{4 items},
               ⊖{
                    "id": "b59bdab8-3560-4cfe-8009-b0461f74c417",
                    "internal": true,
                    "rel": "disruptions",
                    "templated": false
                    "type": "disruption",
                }
            ]
        },
       ⊕{5 items}
    ],
    "display_informations": ⊕{9 items},
    "first_datetime": ⊕{5 items}
    "last_datetime": ⊕{5 items},
    "links": ⊕[4 items],
    "route": ⊕{8 items},
    "stop_point": ⊕{11 items},
```
Call: https://api.sncf.com/v1/coverage/sncf/routes/route%3AOCE%3A46-TrainTER-87746008-87747006/stop_areas/stop_area%3AOCE%3ASA%3A87741009/stop_schedules?from_datetime=20190619T113500&

Author: TO DO by @pbougue

        },
       ⊕{5 items},
    ],
    "display_informations": ⊕{9 items},
    "first_datetime": ⊕{5 items}
    "last_datetime": ⊕{5 items},
    "links": ⊕[4 items],
    "route": ⊕{8 items},
    "stop_point": ⊕{11 items},
}
```

> In the list of "date_times" available in the response, the parameter "data_freshness" is "realtime" and the fileld "departure_date_time" take the delay into account.

Reviewer: ```suggestion
In the list of "date_times" available in the response, the parameter "data_freshness" is "realtime" and the field "date_time" takes the delay into account, whereas "base_date_time" shows the base-schedule departure datetime.
```

> A list of the disruptions impacting the stop schedules is also present at the root level of the response.

Reviewer: You can add a word on `date_times[0].links`

Author: TO DO by @pbougue

## <a name="REDUCED_SERVICE"></a>Reduced service

![image](reduced_service.png)

``` shell

> # Extract of an impacted stop of /disruptions

Reviewer: minor:
```suggestion
# Extract of an impacted stop from /disruptions
```

{
    "arrival_status": "deleted",
    "cause": "",
    "departure_status": "deleted",
    "is_detour": false,
    "stop_point": ⊕{7 items},
    "stop_time_effect": "deleted",
}
```
The effect of the disruption is `REDUCED_SERVICE`. It means that the train won't be serving one or more stations in its journey.

In the disruption, the deleted stations can be found in the list of "impacted_stops" with the departure/arrival status set to "deleted". See the [disruption](#disruption) objects section for its full content and description.

### Journeys

> If the station deleted is the destination of the journey, Navitia will compute an other itinerary to the requested station. If not, the disruption will be present at the root level of the response for information, but it won't be affecting the journey.

Reviewer: ```suggestion
If the stop deleted is the origin/destination of a section of the journey, Navitia will compute a different itinerary from/to the requested station. If not, the disruption will be present at the root level of the response for information, but it won't be affecting the journey.
```

Reviewer: Also, not sure about the behavior when a middle stop is removed : I think there is always a link in display_info to the disruption, and I don't know how the midlle "stop_date_times" is displayed (no time?).
Can be worth a test and adding it to the doc (shell + update the `If not` part) if it's easy.

Author: TO DO by @pbougue with NO response exemple

### Departures & Stop Schedules

> The departure time of the train with the reduced service simply won't be displayed in the list of departures/stop_schedules if "data_freshness" is set to "realtime".

Reviewer: If `data_freshness=base_schedule` then we get the datetime and a link to the disruption I think.

## <a name="MODIFIED_SERVICE"></a>Modified service

> ![image](modified_service.png)

Reviewer: :gb: : There "Ajoutée" is :fr: in image provided.

Author: same same @ArthurLeyv

``` shell
# Extract of an impacted stop of /disruptions
{
    "amended_arrival_time": "185500",
    "amended_departure_time": "190000",
    "arrival_status": "added"
    "cause": "Ajout d'une desserte",
    "departure_status": "added",
    "is_detour": false,
    "stop_point": ⊕{7 items},
    "stop_time_effect": "added",
},
```

> The effect of the disruption is `MODIFIED_SERVICE`. It means that there is one or several stop points added into the journey. This can be at any position in the journey (origin and destination included).

Reviewer: ```suggestion
The effect of the disruption is `MODIFIED_SERVICE`. It means that there is one or several stop points added into the trip. This can be at any position in the trip (origin and destination included).
```

> In the disruption, new stop points can be found in the list of "impacted_stops" with the departure/arrival status set to "added". See the [disruption](#disruption) objects section for its full content and description.

Reviewer: minor: add newline in display
```suggestion
In the disruption, new stop points can be found in the list of "impacted_stops" with the departure/arrival status set to "added".  
See the [disruption](#disruption) objects section for its full content and description.
```
Same on all the "See the [disruption]".

> In the disruption, new stop points can be found in the list of "impacted_stops" with the departure/arrival status set to "added". See the [disruption](#disruption) objects section for its full content and description.

Reviewer: minor: I also think that for SNCF we need to document `is_detour` attribute as they asked for it.
Something like:

If attribute `"is_detour": true` it means that this stops explicitly replaces a deleted one (shown before or after).

Then we should add the same in reduced service section.

Reviewer: Or better: write nothing & just hide it from the example shell (as this part of the interface is not something we want to enforce :wink: )

<div></div>
### Journeys

``` shell
# Request example for /journeys
http://api.navitia.io/v1/coverage/<coverage>/journeys?from=<origin>&to=<destination>&data_freshness=realtime
```

``` shell
# Extract of the public transport section of the response /journeys

    "sections": ⊖[
       ⊖{
            "additional_informations": ⊕[1 item],
            "arrival_date_time": "20190605T204500",
            "base_arrival_date_time": "20190605T204500",
            "co2_emission": ⊕{2 items},
            "data_freshness": "realtime",
            "departure_date_time": "20190605T160000",

>             "display_informations": ⊕{13 items},

Reviewer: Same: it's probably worth expanding display_information.links here (and add a word in description)

Author: TO DO by @pbougue

            "duration": 28560,
            "from": ⊕{5 items},
            "geojson": ⊕{3 items},
            "id": "section_1_0",
            "links": ⊕[6 items],
            "stop_date_times": ⊖[
                ⊕{7 items},
                ⊖{
                     "additional_informations": [],
                     "arrival_date_time": "20190605T185500",
                     "departure_date_time": "20190605T190000",
                     "links": []
                     "stop_point": ⊕{7 items},
                 },
                ⊕{7 items},
            ]
            "to": ⊕{5 items},
            "type": "public_transport",
        },
    ]
```

> The status of the journey is `MODIFIED_SERVICE`. In a public transport section of the response, "arrival_date_time"/"departure_date_time" are the arrival/departure time of a new stop_point. New stop points are only available when the "data_freshness" parameter is set to "realtime". 

Reviewer 3: Same here with "stop_point" ?

Reviewer: Ah! Yeah, would be nice to be consistent :)

> The status of the journey is `MODIFIED_SERVICE`. In a public transport section of the response, "arrival_date_time"/"departure_date_time" are the arrival/departure time of a new stop_point. New stop points are only available when the "data_freshness" parameter is set to "realtime". 

Reviewer: ```suggestion
The status of the journey is `MODIFIED_SERVICE`. In a public transport section of the response, "arrival_date_time"/"departure_date_time" are the arrival/departure times of an added stop_point. New stop points are only used when the "data_freshness" parameter is set to "realtime". 
```

A list of the disruptions impacting the journey is also present at the root level of the response.

> <aside class="notice">

Reviewer: This can be removed (or added also on "reduced service part").

Author: Remove both notices and put it in the top part

For a disruption to be present in the response, the request has to be made during its application period.
</aside>

> <div></div>

Reviewer: This one is useless, no?

                    "base_arrival_date_time": "20190605T193200",
                    "base_departure_date_time": "20190605T193400"
                    "departure_date_time": "20190605T194900",
                    "links": [],
                    "stop_point": ⊕{7 items},
                },
               ⊕{7 items},
               ⊕{7 items},
            ]
            "to": ⊕{5 items},
            "type": "public_transport",
        }
    ]
```

The status of the journey is `SIGNIFICANT_DELAYS`.

In a public transport section of the response:

* "base_arrival_date_time"/"base_departure_date_time" represent the scheduled arrival/departure time without taking into account the delay
* whereas "arrival_date_time"/"departure_date_time" are the actual arrival/departure time, after the delay is applied

The delay can also be observed for every stop point of the journey with the same parameters in "stop_date_times".  
If the parameter "data_freshness" is set to "base_schedule", then "base_arrival_date_time"/"base_departure_date_time" = "arrival_date_time"/"departure_date_time".

A list of the disruptions impacting the journey is also present at the root level of the response.  

> A link to concerned disruption can be found in the section "display_informations".

Author: ```suggestion
A link to the concerned disruption can be found in the section "display_informations".
```

### Journeys

If the stop deleted is the origin/destination of a section of the journey, Navitia will compute a different itinerary to the requested station.  

> If not, the disruption will be present in the `display_informations.links` part of the section for information, but it won't be affecting the journey.

Author: Is it not useful to have an extract from the response to show this? Maybe not because it's already in the previous example of /journeys?!

Also, for consistency, you wrote `display_informations.links` and I wrote 'A link to this disruption can be found in the section “display_informations” '. Let's decide which to way to write it and put it everywhere.
